target_chains/solana/sdk/js/pyth_solana_receiver/src/PythSolanaReceiver.ts

```diff
@@ -39,12 +39,150 @@ import {
+ * await transactionBuilder.withPostPriceUpdates(priceUpdateData);
+ * await transactionBuilder.withPriceConsumerInstructions(...)
+ * transactionBuilder.withCloseInstructions();
+ *
+ * await pythSolanaReceiver.provider.sendAll(await transactionBuilder.getVersionedTransactions({computeUnitPriceMicroLamports:1000000}))
+ * ```
+ */
+export class PythTransactionBuilder extends TransactionBuilder {
+  readonly pythSolanaReceiver: PythSolanaReceiver;
+  private closeInstructions: InstructionWithEphemeralSigners[];
+  private priceFeedIdToPriceUpdateAccount: Record<string, PublicKey>;
+
+  constructor(pythSolanaReceiver: PythSolanaReceiver) {
+    super(pythSolanaReceiver.wallet.publicKey, pythSolanaReceiver.connection);
+    this.pythSolanaReceiver = pythSolanaReceiver;
+    this.closeInstructions = [];
+    this.priceFeedIdToPriceUpdateAccount = {};
+  }
+
+  /**
+   * Add instructions to post price updates to the builder.
+   *
+   * @param priceUpdateDataArray the output of the `@pythnetwork/price-service-client`'s `PriceServiceConnection.getLatestVaas`. This is an array of verifiable price updates.
+   */
+  async withPostPriceUpdates(priceUpdateDataArray: string[]) {
+    const {
+      postInstructions,
+      priceFeedIdToPriceUpdateAccount,
+      closeInstructions,
+    } = await this.pythSolanaReceiver.buildPostPriceUpdateInstructions(
+      priceUpdateDataArray
+    );
+    this.closeInstructions.push(...closeInstructions);
+    this.priceFeedIdToPriceUpdateAccount = {
+      ...this.priceFeedIdToPriceUpdateAccount,
+      ...priceFeedIdToPriceUpdateAccount,
+    };
+    this.addInstructions(postInstructions);
+  }
+
+  /**
+   * Add instructions to post partially verified price updates to the builder.
+   *
+   * @param priceUpdateDataArray the output of the `@pythnetwork/price-service-client`'s `PriceServiceConnection.getLatestVaas`. This is an array of verifiable price updates.
+   *
+   * Partially verified price updates are price updates where not all the guardian signatures have been verified. By default this methods checks `DEFAULT_REDUCED_GUARDIAN_SET_SIZE` signatures when posting the VAA.
+   * If you are a on-chain program developer, make sure you understand the risks of consuming partially verified price updates here: {@link https://github.com/pyth-network/pyth-crosschain/blob/main/target_chains/solana/pyth_solana_receiver_state/src/price_update.rs}.
+   *
+   * @example
+   * ```typescript
+   * const priceUpdateData = await priceServiceConnection.getLatestVaas([
+   *    SOL_PRICE_FEED_ID,
+   *    ETH_PRICE_FEED_ID,
+   * ]);
+   *
+   * const transactionBuilder = pythSolanaReceiver.newTransactionBuilder();
+   * await transactionBuilder.withPostPartiallyVerifiedPriceUpdates(priceUpdateData);
+   * await transactionBuilder.withPriceConsumerInstructions(...)
+   * ...
+   * ```
+   */
+  async withPostPartiallyVerifiedPriceUpdates(priceUpdateDataArray: string[]) {
```

Comment:
very nit: normally `withX` implied that you return it and i chain them together. 
`B.withX().withY().build()`. Maybe `addX` be a name that doesn't imply it.

Reply:
yeah i agree with this. generally with builders there's always a question of "does this method mutate the internal state, or does it return a new builder". The `with` naming suggests that it returns a new builder